backend/internal/handlers/utils.go

```diff
@@ -61,3 +63,66 @@ func handleJSONdecodeError(w http.ResponseWriter, err error) {
 		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
 	}
 }
+
+var (
+	errAuthHeaderNotFound     = errors.New("Authorization header expected but not found")
+	errAuthHeaderNotFormatted = errors.New("Authorization header was improperly formatted")
+)
+
+// Helper function to retreive a token sent in standard "Bearer" format from a request
+// (https://tools.ietf.org/html/rfc6750#page-5). If the request doesn't contain an Authorization
+// header or the Authorization header is improperly formatted, getBearerToken returns "".
+// Handlers generally shouldn't call this function, and should instead call getSessionID or
+// getApiKey (TODO) to specify which type of token they are expecting.
+func getBearerToken(r *http.Request) (string, error) {
+	reqToken := r.Header.Get("Authorization")
+	if reqToken == "" {
+		// Request did not contain an Authorization header
+		return "", errAuthHeaderNotFound
+	}
+	splitToken := strings.Split(reqToken, "Bearer ")
+	if len(splitToken) == 1 {
```

Comment:
nit: this check should be `splitToken := strings.Fields(reqToken); if len(splitToken) != 2 || splitToken[0] != "Bearer" { ...`, to avoid other invalid values like `Bearer foo bar` for example